src/Fixer/Operator/TernaryToElvisOperatorFixer.php

```diff
@@ -0,0 +1,265 @@
+<?php
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Fixer\Operator;
+
+use PhpCsFixer\AbstractFixer;
+use PhpCsFixer\FixerDefinition\CodeSample;
+use PhpCsFixer\FixerDefinition\FixerDefinition;
+use PhpCsFixer\Tokenizer\CT;
+use PhpCsFixer\Tokenizer\Tokens;
+
+/**
+ * @author SpacePossum
+ */
+final class TernaryToElvisOperatorFixer extends AbstractFixer
+{
+    /**
+     * Lower precedence and other valid preceding tokens.
+     *
+     * @var array
+     */
+    private static $validBeforeEndTypes = [
+        '=',
+        [T_AND_EQUAL],    // &=
+        [T_CONCAT_EQUAL], // .=
+        [T_DIV_EQUAL],    // /=
+        [T_DIV_EQUAL],    // =',
+        [T_MINUS_EQUAL],  // -=
+        [T_MOD_EQUAL],    // %=
+        [T_MUL_EQUAL],    // *=
+        [T_OR_EQUAL],     // |=
+        [T_PLUS_EQUAL],   // +=
+        [T_POW_EQUAL],    // **=
+        [T_SL_EQUAL],     // <<=
+        [T_SR_EQUAL],     // >>=
+        [T_XOR_EQUAL],    // ^=
+        '(',
+        '[',
+        '{',
+        '}',
+        ',',
+        ';',
+        [T_OPEN_TAG],
+        [CT::T_ARRAY_INDEX_CURLY_BRACE_OPEN],
+    ];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Use the Elvis operator `?:` where possible.',
+            [
+                new CodeSample(
+                    "<?php\n\$foo = \$foo ? \$foo : 1;\n"
+                ),
+                new CodeSample(
+                    "<?php \$foo = \$bar[a()] ? \$bar[a()] : 1; # \"risky\" sample, \"a()\" only gets called once after fixing\n"
+                ),
+            ],
+            null,
+            'Risky when relying on functions called on both sides of the `?` operator.'
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getPriority()
+    {
+        // should be run before TernaryOperatorSpacesFixer and NoTrailingWhitespaceFixer
```

Comment:
I guess the new priority comments auto review test will fail on this.

Reply:
I hope so! lets see what a rebase does